common4j/src/main/com/microsoft/identity/common/java/opentelemetry/CryptoFactoryTelemetryHelper.java

```diff
@@ -0,0 +1,87 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.opentelemetry;
+
+import static com.microsoft.identity.common.java.opentelemetry.AttributeName.crypto_algorithm;
+import static com.microsoft.identity.common.java.opentelemetry.AttributeName.crypto_controller;
+import static com.microsoft.identity.common.java.opentelemetry.AttributeName.crypto_exception_stack_trace;
+
+import com.microsoft.identity.common.java.crypto.ICryptoFactory;
+import com.microsoft.identity.common.java.exception.ClientException;
+import com.microsoft.identity.common.java.util.StringUtil;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Scope;
+import lombok.NonNull;
+
+public class CryptoFactoryTelemetryHelper {
+
+    private static final String TAG = CryptoFactoryTelemetryHelper.class.getSimpleName();
+
+    private static final Tracer sTracer = GlobalOpenTelemetry.getTracer(TAG);
+
+    private static final String SPAN_PREFIX = "CryptoFactory_";
+
+    /**
+     * A helper class that consolidate all the telemetry emitting work
+     * for crypto operation in one place.
+     *
+     * @param cryptoOperation name of the crypto operation.
+     * @param algorithmName name of the algorithm.
+     * @param cryptoFactory an {@link ICryptoFactory} object.
+     * @param cryptoOperation a callback that wraps around the crypto operation to be performed.
+     * @return result of the crypto operation.
+     * */
+    public static <T> T performCryptoTaskAndUploadTelemetry(@NonNull final CryptoFactoryOperationName operationName,
+                                                            @NonNull final String algorithmName,
+                                                            @NonNull final ICryptoFactory cryptoFactory,
+                                                            @NonNull final ICryptoOperationCallback<T> cryptoOperation)
+            throws ClientException {
+        final Span span = sTracer.spanBuilder(getSpanName(operationName)).startSpan();
+        try (final Scope scope = span.makeCurrent()) {
+            span.setAttribute(crypto_controller.name(), cryptoFactory.getClass().getSimpleName());
+            span.setAttribute(crypto_algorithm.name(), algorithmName);
+            span.setStatus(StatusCode.OK);
```

Comment:
We're setting status "OK" here - what if it fails on next line `cryptoOperation.perform()`? Do we want to mark status ERROR if that fails?

Reply:
for crypto operation, if an exception is not thrown = success.